service/history/queue/timer_queue_processor_base.go

```diff
@@ -203,8 +206,18 @@ processorPumpLoop:
 		case <-t.shutdownCh:
 			break processorPumpLoop
 		case <-t.timerGate.FireChan():
-			if t.redispatchQueue.Len() > t.options.MaxRedispatchQueueSize() {
-				t.redispatchTasks()
+			maxRedispatchQueueSize := t.options.MaxRedispatchQueueSize()
+			if t.redispatcher.Size() > maxRedispatchQueueSize {
+				t.redispatcher.Redispatch(maxRedispatchQueueSize)
+				t.redispatcher.Redispatch(maxRedispatchQueueSize)
```

Comment:
Why do we need to do redispatch twice here?

Reply:
Thanks for catching this! I think it's added by mistake when resolving merge conflicts. Removed.